rust/template/ddlog.h

```diff
@@ -232,12 +233,38 @@ extern int ddlog_transaction_commit(ddlog_prog hprog);
  * This function will fail if there is no transaction in progress.
  */
 extern int
-ddlog_transaction_commit_dump_changes(ddlog_prog hprog,
-                                      void (*cb)(void *arg,
-                                                 table_id table,
-                                                 const ddlog_record *rec,
-                                                 bool polarity),
-                                      uintptr_t cb_arg);
+ddlog_transaction_commit_dump_changes(
+        ddlog_prog hprog,
+        void (*cb)(void *arg,
+                   table_id table,
+                   const ddlog_record *rec,
+                   bool polarity),
+        uintptr_t cb_arg);
+
+/*
+ * Same as `ddlog_transaction_commit_dump_changes`, but serializes changes to a
+ * FlatBuffer.  On success, returns pointer to FlatBuffer, size and capacity of
+ * the buffer, and offset where valid data starts inside the buffer.
+ */
+extern int
+ddlog_transaction_commit_dump_changes_to_flatbuf(
```

Comment:
Why would the offset be not zero?
You should probably say that the buffer has to be deallocated by ddlog_flatbuf_free.

Reply:
It's just how FlatBuffers work. The offset is almost never zero. Not sure why.